packages/appium/lib/appium.js

```diff
@@ -492,6 +486,20 @@ class AppiumDriver extends BaseDriver {
       // if we make it here, we know that the default behavior is handled
       cmdHandledBy.default = true;
 
+      if (reqForProxy) {
+        // we would have proxied this command had a plugin not handled it, so the default behavior
+        // is to do the proxy and retrieve the result internally so it can be passed to the plugin
+        // in case it calls 'await next()'. This requires that the driver have defined
+        // 'proxyCommand' and not just 'proxyReqRes'.
+        if (!dstSession.proxyCommand) {
+          throw new Error(`The default behavior for this command was to proxy, but the driver ` +
```

Comment:
I would love it if we'd have specific subclasses of `Error`, each with their own unique `code` property. Never too late to start!!

Reply:
Also... this is rather verbose.  Maybe in the future we could provide a URL to view detailed explanations about each error (indexed by `code`, of course).

Reply:
that's a good idea. i think that would best be its own project of just grepping the code and systematically creating these things. in general i don't mind a bit of verbosity in my error messages, they also serve as good context for developers in the code.